legacy/reqcounter/reqcounter_test.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reqcounter_test
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	rc "sigs.k8s.io/k8s-container-image-promoter/legacy/reqcounter"
+)
+
+// defaultTime should be used as a timestamp for all RequestCounter objects.
+// The actual time represents: September 22, 2002 at 05:03:16.
+var defaultTime, _ = time.Parse("020106 150405", "020106 150405")
+
+// NewRequestCounter returns a RequestCounter with the given number of requests.
+// All other object fields are set to default values.
+func NewRequestCounter(requests uint64) rc.RequestCounter {
+	return rc.RequestCounter{
+		Mutex:    sync.Mutex{},
+		Requests: requests,
+		Since:    defaultTime,
+		Interval: time.Second,
+	}
+}
+
+func TestInit(t *testing.T) {
+	rc.Init()
+	// Ensure request counting is enabled.
+	require.True(t, rc.EnableCounting, "Init did not enable counting.")
+	// Ensure counters are created.
+	require.Greater(t, len(rc.Monitor.Counters), 0, "Init did not create any request counters within the global Monitor.")
+	// Ensure at least one counter uses the MeasurementWindow.
+	found := false
+	for _, counter := range rc.Monitor.Counters {
+		if counter.Interval == rc.MeasurementWindow {
+			found = true
+			break
+		}
+	}
+	require.True(t, found, "No request counters are using the Interval: MeasurementWindow.")
+}
+
+func TestIncrement(t *testing.T) {
+	// Create counters which use these request counts and
+	// populate the Monitor global variable.
+	counters := []rc.RequestCounter{
```

Comment:
If you do

```
type RequestCounters []RequestCounter
```

in the reqcounter package, it would make this line a little easier to read, because it would become

```
counters := rc.RequestCounters{
```
```

it would make this a

Reply:
I still can't remove the local definition of each counter. This is because the counters must be passed as pointers. Regardless, I will still make this change.